index.js

```diff
@@ -7,28 +7,40 @@ const apiVersion = process.env.APIVERSION;
 
 const processingDate = new Date().toUTCString();
 
-exports.handler = function(event, context, callback) {
-  const payload = event.Records[0].Sns;  
-  const index = `${prefix}`;
-  const contentLength = Buffer.byteLength(JSON.stringify(payload), 'utf8');
-  const stringToSign = 'POST\n' + contentLength + '\napplication/json\nx-ms-date:' + processingDate + '\n/api/logs';
-  const signature = crypto.createHmac('sha256', new Buffer.from(sharedKey, 'base64')).update(stringToSign, 'utf-8').digest('base64');
-  const authorization = 'SharedKey ' + workspaceId + ':' + signature;
-  const headers = {
-    "content-type": "application/json",
-    "Authorization": authorization,
-    "Log-Type": index,
-    "x-ms-date": processingDate
-  };
+exports.handler = async (event, context) => {
+  return new Promise(async (resolve, reject) => {
+    const payload = event.Records[0].Sns;
+    const index = `${prefix}`;
+    const contentLength = Buffer.byteLength(JSON.stringify(payload), 'utf8');
+    const stringToSign = 'POST\n' + contentLength + '\napplication/json\nx-ms-date:' + processingDate + '\n/api/logs';
+    const signature = crypto.createHmac('sha256', new Buffer.from(sharedKey, 'base64')).update(stringToSign, 'utf-8').digest('base64');
+    const authorization = 'SharedKey ' + workspaceId + ':' + signature;
+    const headers = {
+      "content-type": "application/json",
+      "Authorization": authorization,
+      "Log-Type": index,
+      "x-ms-date": processingDate
+    };
 
-  request
-  .post(`https://${workspaceId}.ods.opinsights.azure.com/api/logs?api-version=${apiVersion}`)
-  .send(payload)
-  .set(headers)
-  .end((err,res) => {
-    if (err) {        
-      return err;
-    }
-	return res;
+    await request
+      .post(`https://${workspaceId}.ods.opinsights.azure.com/api/logs?api-version=${apiVersion}`)
+      .send(payload)
+      .set(headers)
+      .end((err, res) => {
+        if (err) {
+          reject({
+            'statusCode': err.statusCode,
+            'body': JSON.stringify({
+              message: err,
+            })
+          });
+        }
+        resolve({
+          'statusCode': res.statusCode,
+          'body': JSON.stringify({
+            message: res,
+          })
+        });
+      });
```

Comment:
According to the superagent doc, this is not how you _await_ for a request
```suggestion
return request
  .post(
    `https://${workspaceId}.ods.opinsights.azure.com/api/logs?api-version=${apiVersion}`
  )
  .send(payload)
  .set(headers);
```
`end()` doesn't need to be used when making an async call, success/rejection should be handled in lambda in this case (need to test)

Reply:
Removed the end() now and also added try and catch block to handle the exception. Also tested on sam local cli and it is working fine.

Reply:
have you tried my suggestion ?

Reply:
Yes, I have tried the same way